climada/hazard/base.py

```diff
@@ -719,6 +723,14 @@ def select(self, event_names=None, date=None, orig=None, reg_id=None, reset_freq
             if not np.any(sel_cen):
                 LOGGER.info('No hazard centroids with region %s.', str(reg_id))
                 return None
+        if extent is not None:
+            cent_ext = self.centroids.select(extent=extent)
+            cent_ext_view = cent_ext.coord.view(dtype='float64,float64').reshape(-1)
+            cent_view = self.centroids.coord.view(dtype='float64,float64').reshape(-1)
+            sel_cen &= np.isin(cent_view, cent_ext_view)
```

Comment:
Instead of trying to reproduce the selection of `Centroids.select`, I suggest to add a new method `Centroids.select_mask` that implements the lines https://github.com/CLIMADA-project/climada_python/blob/1d8da77b06e72dc095f3d00b9e22c88ad8e6635a/climada/hazard/centroids/centr.py#L963-L974. It returns the `sel_cen` mask from given `reg_id` and/or `extent` parameters. This function should be used in `Centroids.select` and it should be used here (in `Hazard.select`) to compute the `sel_cen` mask to be used below.

Reply:
That is a good idea.

Reply:
done here 011f307